### PR TITLE
E2E tests for volume live migration

### PR DESCRIPTION
**Pull request by joshimoo**
longhorn/longhorn#2127

depends on https://github.com/longhorn/longhorn-manager/pull/822

### CODE REVIEW

File: manager/integration/tests/test_migration.py [hunk 1]
```diff
@@ -15,38 +14,29 @@ def test_migration_confirm(clients, volume_name):  # NOQA
 
 
 def migration_confirm_test(clients, volume_name, base_image=""):  # NOQA
```
Comment (shuo-wu) [inline manager/integration/tests/test_migration.py:16]: Need the test skeleton. Same below

File: manager/integration/tests/test_migration.py [hunk 2]
```diff
@@ -56,31 +46,80 @@ def test_migration_rollback(clients, volume_name):  # NOQA
...
+    volume = client.by_id_volume(volume_name)
+    volume.attach(hostId=common.get_self_host_id())
+    volume = common.wait_for_volume_healthy(client, volume_name)
+    common.check_volume_data(volume, data)
+
+    volume.detach(hostId="")
+    volume = common.wait_for_volume_detached(client, volume_name)
+
+
+def setup_migration_test(clients, volume_name, base_image=""): # NOQA
```
Comment (shuo-wu) [inline manager/integration/tests/test_migration.py:105]: Rename `base_image` to `backing_image`

File: manager/integration/tests/test_migration.py [hunk 3]
```diff
@@ -56,31 +46,80 @@ def test_migration_rollback(clients, volume_name):  # NOQA
...
+    """
+    Creates a new migratable volume then attaches it to the
+    current node to write some test data on it.
+    """
+    client = get_random_client(clients)
+    volume = client.create_volume(name=volume_name, size=SIZE,
+                                  numberOfReplicas=REPLICA_COUNT,
+                                  baseImage=base_image,
```
Comment (shuo-wu) [inline manager/integration/tests/test_migration.py:113]: Rename `base_image` to `backing_image`

File: manager/integration/tests/test_migration.py [hunk 4]
```diff
@@ -15,38 +14,29 @@ def test_migration_confirm(clients, volume_name):  # NOQA
...
 
-    volume = volume.migrationStart(nodeId=host2)
-    attached_nodes = get_volume_attached_nodes(volume)
-    assert host1 in attached_nodes
-    assert volume["migrationNodeID"] == host2
+    volume.attach(hostId=host2)
     with pytest.raises(Exception) as e:
-        volume.migrationConfirm()
+        volume.detach(hostId=host1)
     assert "migration is not ready" in str(e.value)
```
Comment (shuo-wu) [inline manager/integration/tests/test_migration.py]: Is this for checking a transient state? If YES, we need to remove this. Same below

**Review by shuo-wu — CHANGES_REQUESTED** [review]

**Review by shuo-wu — CHANGES_REQUESTED** [review]
The content LGTM. But the modification based on the review should be in the 1st commit.

**Comment by joshimoo** [comment]
squashed the commits.

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**